### DISCUSSION TITLE
POCSAG SI Radio

### DISCUSSION THREAD

**Discussion by TurfControl** [jgromes/RadioLib#1176] [Q&A]
I have a project that is a rx board that uses POCSAG 512.  i designed a "Breakout" board around the SI4362 module and got everything working in direct RX mode writing my own modules (i was unaware of this at the time).  placing my radio onto the "decoder" board that i was replacing the pager on did work as expected. the SI4362 demodulated the POCSAG signals as expected and the on board chip worked.  The next step in my design is to decode the POCSAG on my micrfocontroller and even include a display screen. I first got this working with a SAMD21 in Arduino. i have had a very hard time designing my own POCSAG decoder protocal, as i always used the direct RX but couldnt get the packet handling to work in an expected fashion.  Looking at the SI4432, the 4362 is a part of the same family and pin for pin compatible according to the datasheet.  The API seems to be nearly identical as well.

Now to my question:
Does anyone feel that there would be a way to implement radiolib onto the 4362?  

i cannot find a location of the Patch file that is required after a Power on reset, so i am unable to update the patch.  I have been looking deeop into the library in an attempt to write a module for the 4362 and just piggy back onto the POCSAG portion that was written but i am unable to really get the class' to align with the API as i would expect them to.

Second question:
is there any help someone can give me towards decoding the POCSAG on my own if utilizing the library is impossible?

more about my system:
i have a basestation People Finder II that uses POCSAG 512 to send out commands to these "decoder" boards.  the payload is always a numerical message, and i have been able to decode it with PDW accurately. People finder and decoders are UHF nbarrowband in the frequency range of 450-470MHz. 

EDIT:

I will gladly supply the GERBERS / BOM or even source files for anyone that would like a breakout boad of this. There are other Radios in the same family that "SHOULD" work with this layout in its entirety, however i did not include anything for TX.  the Antenna uses an LNA matching as recomended by the datasheet using recomended parts (almost all available for pick and place from JLCPCB).  the only parts required to place by hand are the pin headers,  the TCXO and the Radio itself.

**Comment by jgromes** [comment]
> Does anyone feel that there would be a way to implement radiolib onto the 4362?

Definitely, it will most likely require writing a new driver for it. Because of RadioLib's internal modular design, the existing Si4432 could be used as the base to make it easier. There were also some issues discussing support for other SI radio modules (#920 and #655), though I don't know whether there has been any progress.

> i cannot find a location of the Patch file that is required after a Power on reset

I don't understand what you mean by that, I'm not really familiar with Si446x devices.

> is there any help someone can give me towards decoding the POCSAG

You can start by examining RadioLib implemetation, specifically `PagerClient::startReceive` and `PagerClient::readData`.

**Reply by TurfControl** [reply]
it was a PITA to find truly, and there are 2 different files.  using WDS to create a base config file and choosing to export for 3rd party exports 100% of the commands normally written for the SI MCU's.  after saving that file somewhere, it is under ...\src\drivers\radio\Si446x  (even though i used a 4362 it uses this folder as the API is 100% identical for these radios cant speak for the 443x yet).  it is labled as just si446x_patch.h and depending on what chip is selected within the WDS will determine the patch supplied.  i have both B rev and C rev IC's and got them to work using a basic sketch not a complete driver.

thank you for the direction for POCSAG i appreciate it, and the work that has been done here truly is quite impressive!
